controller/iteration.go

```diff
@@ -213,10 +213,60 @@ func (c *IterationController) Update(ctx *app.UpdateIterationContext) error {
 		if ctx.Payload.Data.Attributes.UserActive != nil {
 			itr.UserActive = *ctx.Payload.Data.Attributes.UserActive
 		}
+		var oldSubtree []iteration.Iteration
+		if ctx.Payload.Data.Relationships != nil && ctx.Payload.Data.Relationships.Parent != nil {
+			// update parent of Iteration
+			// do not allow root-iteraiton to update its parent
+			if itr.IsRoot(itr.SpaceID) {
+				return jsonapi.JSONErrorResponse(ctx,
+					errors.NewForbiddenError("Parent of root iteration can not be updated"))
+			}
+			newParentID := ctx.Payload.Data.Relationships.Parent.Data.ID
+			if newParentID == nil {
+				return jsonapi.JSONErrorResponse(ctx,
+					errors.NewBadParameterError("Data.Relationships.Parent.ID", newParentID).Expected("not nil"))
+			}
+			pid, err := uuid.FromString(*newParentID)
+			if err != nil {
+				return jsonapi.JSONErrorResponse(ctx, errors.NewBadParameterError("Data.Relationships.Parent.ID", newParentID))
+			}
+			// Iteration itself can not be parent of self
```

Comment:
@pranavgore09 There is one more case that should be checked. The new Parent Iteration ID should not belong to one of the child of the subject Iteration. Otherwise it will form a cycle.

Reply:
thanks @DhritiShikhar added that case in here https://github.com/fabric8-services/fabric8-wit/pull/1770/commits/f93d45b91c175a32505dcb354c4faf9cbcec51cb

Reply:
@kwk can you also please take a look again since code is updated https://github.com/fabric8-services/fabric8-wit/pull/1770/commits/f93d45b91c175a32505dcb354c4faf9cbcec51cb